nss-day-cohort-41/tabloidmvc-oysters-rockefeller-rose
# Cj comments

# Description
Developed a Comments feature that allows users to view and create comments on a given Post. 

The feature is currently built for admin user types. Admins can view all comments for a post in descending order by date, create new comments, delete any comments, and edit their own comments.

Future updates will refactor this feature. When the program supports the "Author" user type, Comments will allow that user type to delete only their own comments

# Ticket Reference
> [#19 - View Post Comments](https://github.com/nss-day-cohort-41/tabloidmvc-oysters-rockefeller-rose/issues/19)

> [#20 - Create a Comment](https://github.com/nss-day-cohort-41/tabloidmvc-oysters-rockefeller-rose/issues/20)


## Type of change


# Testing Instructions
- Run both files in the SQL folder.
- Have at least 2 posts created via SQL
- Have at least 3 comments created via SQL, two of which with a PostId of 1, and the third with a PostId of 2.
- Start program with ISS Express
- Go to localhost:44336 in your browser.
- Login with "admin@example.com"
- Navigate to the Posts link in the navbar.
- Click the eyeball button on the first Post, which should have a title "C# is the Best Language"
- Scroll to the bottom of the page and click the "View Comments" button.
- Visually confirm the Post title displayed directly beneath the navabar is the correct Post title.
- Visually confirm each listing contains the subject, content, author's display name, and creation date (MM/DD/YYYY).
- Visually confirm only the two comments with an ID of 1 are displaying, in order by most recent date, by starting a query in the TabloidMVC database with command SELECT * FROM Posts
- Click on the "Add Comment" button below the Post title.
- First, click the "Back to Post" link and visually confirm that the correct post title and all it's comments are displaying.
- Again, click on the "Add Comment" button below the Post title.
- Fill out the fields for subject and content, then click Create.
- Visually confirm that the correct post title and all it's comments are displaying, including the new one that you just created.
- Go back to Visual Studio and start a new SQL Query on the TabloidMVC database with command SELECT * FROM Posts
- Visually confirm that the new comment with an ID of 1 is displaying, in order by most recent date.
- Back in the browser, click the Back to post link at the bottom of the page to return back to the post.

# Checklist:
